verilog/tools/kythe/kythe_facts_extractor.cc

```diff
@@ -26,20 +26,17 @@ namespace verilog {
 namespace kythe {
 
 void KytheFactsExtractor::ExtractKytheFacts(const IndexingFactNode& root) {
-  CreatePackageScopes(root);
-  IndexingFactNodeTagResolver(root);
-}
-
-void KytheFactsExtractor::CreatePackageScopes(const IndexingFactNode& root) {
-  // Searches for packages as a first pass and saves their scope to be used for
-  // imports.
-  for (const IndexingFactNode& child : root.Children()) {
-    if (child.Value().GetIndexingFactType() != IndexingFactType::kPackage) {
-      continue;
-    }
-
-    VName package_vname = ExtractPackageDeclaration(child);
-    Visit(child, package_vname);
+  std::size_t number_of_extracted_facts = 0;
+  do {
+    number_of_extracted_facts = facts_.size();
+    IndexingFactNodeTagResolver(root);
+  } while (number_of_extracted_facts != facts_.size());
+
+  for (const Fact& fact : facts_) {
```

Comment:
Related to a suggestion from a while ago.  Consider isolating the collection of facts from printing, for example, accepting a listener function that may choose to collect or call some other function(s) on facts/edges.  For example a caller may wish to do some alternative sorting on facts/edges (than the ordering you've established).

No change needed for this PR.